src/classmentors/components/cohorts/cohorts.js

```diff
@@ -1036,31 +925,113 @@ function ClmCohortRankPageCtrl($q, $scope, $log, firebaseApp, $firebaseObject, $
                         oneEventData.title = result.title;
                         self.cohortTotalParticipants += oneEventData.participants.length;
                         oneEventData.id = result.$id;
-                        self.cohortEventData.push(oneEventData);
-                        iter++;
-                        loopDBEvents();
+                        oneEventData.userRanks = [];
+                        var userIndex = 0;
+                        loadUserAchievements();
+                        function loadUserAchievements() {
+                            if(userIndex < oneEventData.participants.length) {
+                                var participantRankingRef = db.ref(`classMentors/userProfiles/${oneEventData.participants[userIndex].$id}`);
+                                var participantRankingObj = $firebaseObject(participantRankingRef);
+                                participantRankingObj.$loaded().then(function () {
+                                    var rankingResult = participantRankingObj;
+                                    console.log(rankingResult);
+                                    if(rankingResult.services && rankingResult.services.freeCodeCamp) {
+                                        oneEventData.userRanks.push({"user": rankingResult.user, "total": parseInt(rankingResult.services.freeCodeCamp.totalAchievements)});
+                                    } else {
+                                        oneEventData.userRanks.push({"user": rankingResult.user, "total": 0});
+                                    }
+                                    userIndex++;
+                                    loadUserAchievements();
```

Comment:
No particular reason, I just wanted to ensure that the program does not attempt to make another call to the firebase until the previous promise had been resolved and the data handled. This seemed like the easiest way to do so. If there are performance of reliability related reasons NOT to use this method, then I can also change it.